corehq/apps/hqwebapp/signals.py

```diff
@@ -0,0 +1,32 @@
+from datetime import date
+
+from django.contrib.auth.signals import user_logged_in, user_login_failed
+from django.dispatch import receiver
+
+from corehq.apps.users.models import CouchUser
+
+
+def clear_login_attempts(user):
+    user = CouchUser.from_django_user(user)
+    if user.is_web_user() and user.login_attempts > 0:
+        user.login_attempts = 0
+        user.save()
+
+
+@receiver(user_logged_in)
+def clear_failed_logins_and_unlock_account(sender, request, user, **kwargs):
+    clear_login_attempts(user)
+
+
+@receiver(user_login_failed)
```

Comment:
As far as I can tell, the only other way it could fail is with an incorrect username, but that will not affect lockout since the failed attempt will not be able to be linked to an account. Is there something you were thinking of specifically that this could catch by mistake?